runtime/jcl/common/reflecthelp.c

```diff
@@ -829,6 +829,13 @@ createField(struct J9VMThread *vmThread, jfieldID fieldID)
 	J9VMJAVALANGREFLECTFIELD_SET_DECLARINGCLASS(vmThread, fieldObject, J9VM_J9CLASS_TO_HEAPCLASS(j9FieldID->declaringClass));
 #if defined(USE_SUN_REFLECT)
 	J9VMJAVALANGREFLECTFIELD_SET_MODIFIERS(vmThread, fieldObject, j9FieldID->field->modifiers & CFR_FIELD_ACCESS_MASK);
+#if JAVA_SPEC_VERSION >= 15
+	if (J9_ARE_ANY_BITS_SET(j9FieldID->field->modifiers, J9AccFinal)
+		&& J9_ARE_ANY_BITS_SET(j9FieldID->field->modifiers, J9AccStatic)
```

Comment:
The ANY bit test macro is defined as:
```
#define OMR_ARE_ANY_BITS_SET(value, bits) (0 != ((value) & (bits)))
```
which means this can be a single test.  Though, I think it should be ALL_BITS, not ANY:
```
#define OMR_ARE_ALL_BITS_SET(value, bits) ((bits) == ((value) & (bits)))
```
As we want to test for both FINAL and STATIC:
```suggestion
	if (J9_ARE_ALL_BITS_SET(j9FieldID->field->modifiers, J9AccFinal | J9AccStatic)
```

Reply:
Nice! fixed accordingly.
Also verified that this works for FINAL & STATIC along with combination of other modifiers such as public/private.